compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -925,11 +925,11 @@ private boolean populateRecordFields(BLangExpression expr, BConstantSymbol const
                     return false;
                 }
                 keyValuePair.setBType(newType);
-                if (newType.getKind() != TypeKind.FINITE) {
+                TypeKind kind = newType.getKind();
+                if (kind != TypeKind.FINITE) {
                     constValueMap.get(key).type = newType;
-                    if (newType.getKind() == TypeKind.INTERSECTION) {
-                        exprValueField.setBType(((BIntersectionType) newType).effectiveType);
-                    }
+                    BType type = kind == TypeKind.INTERSECTION ? ((BIntersectionType) newType).effectiveType : newType;
```

Comment:
Is this logic handles the Refered Types?

Reply:
I didn't get. What do you mean by handle referred type? Here, `newType` is the immutable type that is generated for the record field type.

Reply:
I believe Sasindu meant something like

```ballerina
public type AnnotationRecord record {|
    string summary?;
    ExamplesReference examples?;
|};

type ExamplesReference Examples;

public type Examples record {|
    map<ExampleItem> response;
|};
```

This seems to be working, but I would add a test.

Reply:
Added test case